File: Python/Rock, Paper, Scissors.py
# Rock, Paper, Scissors
# Import random first

import random

def play():
    # Welcome message + Create variables
    print("Welcome to Rock, Paper, Scissors game")
    print("""Enter "Exit" to quit the game.""")
    moves = ["Rock", "Paper", "Scissors"]
    win = 0
    loss = 0
    tie = 0
    round = 0

    while True:
        player_move = str(input("""
        Enter [Rock] [Paper] [Scissors] : """))

        if player_move == "Exit":
            print("""
            Scoreboard: """)
            print("Total round = ", round)
            print("Win = ", win)
            print("Loss = ", loss)
            print("Tie = ", tie)
            print("Win rate =", (win/(round))*100 )
            print("Thanks for playing!")
            break

        comp_move = random.choice(moves)
        print("Player's Move: ", player_move)
        print("Computer's Move: ", comp_move)

        if player_move == comp_move:
            tie += 1
            round += 1
            print("Tie")
        elif player_move == "Rock" and comp_move == "Scissors":
            win += 1
            round += 1
            print("You win!")
        elif player_move == "Paper" and comp_move == "Rock":
            win += 1
            round += 1
            print("You win!")
        elif player_move == "Scissors" and comp_move == "Paper":
            win += 1
            round += 1
            print("You win!")
        else:
            loss += 1
            round += 1
            print("It's your Loss")

File: R/Rock, Paper, Scissors.R
## HW02: Create Rock, Paper, Scissor games and exit with score count.

## Create counter and moves options.
moves <- c("Rock", "Paper", "Scissors", "Quit")
win <- 0
tie <- 0
loss <- 0 
round <- 0

## While loop
while (TRUE) {
  player_move <- as.numeric(readline("Choose: Rock[1], Paper[2],Scissors[3], Quit[4]"))
  if(player_move == 4) {
    print("Good bye.")
    break
  }

## Sample moves
  player_move <- moves[player_move]
  computer_move <- moves[sample(1:3, 1)]  
  
## Create rules.
  if(player_move == computer_move) {
    tie <- tie+1
  } else if(player_move == "Rock" & computer_move == "Scissors") {
    win <- win+1
  } else if(player_move == "Paper" & computer_move == "Rock") {
    win <- win+1
  } else if(player_move == "Scissors" & computer_move == "Paper") {
    win <- win+1
  } else {
    loss <- loss+1
  }
  
## Print game visual.
  round <- round+1
  print(paste("Round ----- ", round))
  print(paste("Player move: ", player_move))
  print(paste("Comp move: ", computer_move))
  print(paste("Score: ", "Win =", win ,"Tie =", tie, "Loss =", loss))
}